### PR TITLE
Python operators

### PR DESCRIPTION
**Pull request by maximiliankaul**
This PR implements the new operator overloading functionality for the Python language.

### CODE REVIEW

**Comment by maximiliankaul** [comment]
There are still some problems to discuss:
- some operators like `+` can map to different functions depending on them being used as a unary `__pos__` or binary `__add__` operator
- do we want to include "operators" like `abs()`?
- the documentation say, that `[]` or `()` "roughly" translates to ... -> I'd still hard code them as an exact match

**Comment by oxisto** [comment]
> There are still some problems to discuss:
> 
> * some operators like `+` can map to different functions depending on them being used as a unary `__pos__` or binary `__add__` operator

Unfortunately, I have no good idea how to handle this :( The only "correct" way to do that would be to use a Pair of an class and operator code as a key, something like
```kotlin
Pair(BinaryOperator::class, "+") to "__add__",
Pair(UnaryOperator::class, "+") to "__pos__",
```

which is not too bad. If that makes sense I can prepare this in the base PR

> * do we want to include "operators" like `abs()`?

No

> * the documentation say, that `[]` or `()` "roughly" translates to ... -> I'd still hard code them as an exact match

Yes, that's ok

**Comment by maximiliankaul** [comment]
> > There are still some problems to discuss:
> > 
> > * some operators like `+` can map to different functions depending on them being used as a unary `__pos__` or binary `__add__` operator
> 
> Unfortunately, I have no good idea how to handle this :( The only "correct" way to do that would be to use a Pair of an class and operator code as a key, something like
> 
> ```kotlin
> Pair(BinaryOperator::class, "+") to "__add__",
> Pair(UnaryOperator::class, "+") to "__pos__",
> ```
> 
> which is not too bad. If that makes sense I can prepare this in the base PR

I'm in favor of this approach.

> > * do we want to include "operators" like `abs()`?
> 
> No

I'll remove the comments.
 
> > * the documentation say, that `[]` or `()` "roughly" translates to ... -> I'd still hard code them as an exact match
> 
> Yes, that's ok

:+1:

File: cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonLanguage.kt [hunk 1]
```diff
@@ -49,6 +55,54 @@ class PythonLanguage : Language<PythonLanguageFrontend>(), HasShortCircuitOperat
     override val compoundAssignmentOperators =
         setOf("+=", "-=", "*=", "**=", "/=", "//=", "%=", "<<=", ">>=", "&=", "|=", "^=", "@=")
 
+    // https://docs.python.org/3/reference/datamodel.html#special-method-names
+    override val overloadedOperatorNames:
```
Comment (oxisto) [inline cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonLanguage.kt:59]: Need to add `@Transient`

**Review by oxisto — APPROVED** [review]

**Pull request merged**